### DISCUSSION TITLE
Replacement for getIntialProps in custom _app

### DISCUSSION THREAD

**Discussion by borasemiz** [vercel/next.js#18979] [Help]
I am wondering if there is anything that I can use in place of `getInitialProps` in a next app which has custom `_app`. I saw in [this documentation](https://nextjs.org/docs/api-reference/data-fetching/getInitialProps) that `getInitialProps` is not recommended. There is a suggestion to use either `getStaticProps` or `getServerSideProps`.

But there is a problem with `getServerSideProps`. In [this documentation](https://nextjs.org/docs/basic-features/data-fetching#only-allowed-in-a-page-2), it says that `getServerSideProps` can only be exported from pages. I believe a custom `_app` isn't counted as a page (?).

Indeed, I tried exporting a `getServerSideProps` function from the app module, it doesn't work. The function is not even executed. But, `getInitialProps` gets executed, if there is any. Even in the code example [here](https://nextjs.org/docs/advanced-features/custom-app), `getInitialProps` is shown that it is meant to be used.

So, `getInitialProps` is deprecated, but I cannot use `getServerSideProps`, but I really need one of these functions to work because my App component logic depends on these. What else can I use in place of these two?

**Comment by tunesmith** [comment]
I think you're either mistaken about `getServerSideProps`, or I'm misunderstanding the question. 

When visiting a page, it will by definition load `_app`, and you can't load `_app` without visiting a page. It is true that `getServerSideProps` is not supported in `_app`, but anything you do in `getInitialProps` on a custom `_app`, you should be able to do in `getServerSideProps` on the page level.

So, you cannot use `getServerSideProps` inside an `_app`, but you can remove `getInitialProps` from `_app` and put the logic inside of a page's `getServerSideProps`. If it's something multiple pages need to do, put it in a shared library.

**Comment by satanworker** [comment]
Hey! I've just stumbled upon the same problem, I'm fetching some data on `_app` level and then pass it directly to `<Compoent>` as props to avoid fetching the same data on each page, for example menu structure and contacts, that's impossible to do right now with `getServerSideProps` and if I try to leave `getInitialValues` for  `_app` and use `getServerSideProps` everywhere else I'm getting all the data from `_app` with each request. So seems it's impossible to do something similar with `getServerSideProps`

**Comment by controversial** [comment]
See #10949 (discussion) and #12434 (issue). There hasn't been any word from the maintainers on this issue in a very long time, though.
